Data/Yaml.hs

```diff
@@ -190,6 +195,14 @@ decodeEither' = either Left (either (Left . AesonException) Right)
               . decodeHelper
               . Y.decode
 
+-- | A version of 'decodeEither'' lifted to MonadThrow
+decodeThrow :: (MonadThrow m, FromJSON a) => ByteString -> m a
+decodeThrow = either throwM return . decodeEither'
+
+-- | A version of 'decodeFileEither' lifted to MonadIO, MonadThrow
+decodeFileThrow :: (MonadIO m, MonadThrow m, FromJSON a) => FilePath -> m a
+decodeFileThrow = liftIO . decodeFileEither >=> either throwM return
+
```

Comment:
It's probably worth deprecating at this point some of the decode functions with less useful exception output. I can do that myself after merging this PR, or if you'd like to add some deprecations, go for it.